src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -143,12 +122,35 @@ export class DiscoverPlugin implements Plugin<DiscoverSetup, DiscoverStart> {
       if (this.servicesInitialized) {
         return;
       }
-      const services = await buildServices(core, plugins, this.docViewsRegistry!);
+      const services = await buildServices(core, plugins);
       setServices(services);
       this.servicesInitialized = true;
     };
 
+    this.injectorEmbeddable = await this.getInjectorEmbeddable(core, plugins);
+    const docViewsRegistry = new DocViewsRegistry(this.injectorEmbeddable);
```

Comment:
@flash1293 
Since initializing of an inner angular requires start deps, I had to move docViewsRegistry creation to `start` method. And now the Discover plugin exposes `addDocView` from its **start** interface. I hope such movement is allowed.

Reply:
I would like to keep the registry in `setup` if it's possible. Also with this setup it seems like the inner angular is always initialized in the start phase (even if the discover app isn't / won't be used) - we should avoid that and only initialize if discover (or the saved search embeddable) is used.

Suggestion:
* Move registry back to setup
* Don't pass down the injectorEmbeddable (side note: I would call it `embeddableInjector`), but a function `getInjector: () => Promise<Injector>`
* This function uses `coreSetup.getStartServices` (that's a new thing) to wait for the start services within the setup phase - once that resolves, it will initialize the angular and return it.
* Additionally this function stores the injector object as a private member of the plugin instance, and if it's called again, it returns the already created injector.
* The same function should be used in `registerEmbeddable` to avoid creating to separate angular instances for docs views and saved searches embeddables (they should just re-use the same one)

LTM when somethings unclear.